docs/segfault.md

```diff
@@ -0,0 +1,48 @@
+# How to troubleshoot a segfault
+
+In this document will outline how to configure your system to provide enough information to debug lsif-clang in the case of a segfault. 
+
+## Info to be provided
+
+Along with the info gathered by reading the section below, please also provide:
+
+1. lsif-clang version (commit/tag/etc)
+1. Output of `ldd <path to lsif-clang>`
+1. lsif-clang source (self-compiled/docker/prebuilt binary)
+    - In the case of self-compiled, it may be useful to attach the binary in the bug report
+1. One or both of the below (stacktrace/core dump)
+
+## Stacktrace
+
+The lsif-clang indexer includes the ability to emit a detailed stacktrace (when possible) as shown in the below screenshot if a segfault is encountered. In the case of the stacktrace not being annotated with source fragments as shown below, we may also ask for a core dump. To shorten the feedback loop, it can be beneficial to include the core dump with the original bug report.
```

Comment:
```suggestion
The lsif-clang indexer has the ability to emit a detailed stacktrace (when possible) as shown in the screenshot below if a segfault is encountered. In the case of the stacktrace not being annotated with source fragments as shown below, we may also ask for a core dump. To shorten the feedback loop, it can be beneficial to include the core dump with the original bug report.
```